datacube/api/core.py

```diff
@@ -1034,6 +1041,8 @@ def _make_dask_array(chunked_srcs,
     empties = {}  # type Dict[Tuple[int,int], str]
 
     def _mk_empty(shape: Tuple[int, ...]) -> str:
+        if type(shape) is not tuple:
```

Comment:
If this method accepts something other than Tuple you should update the typehint to reflect.

Presumably something like `Union[Tuple[int, ...], OtherType]`